meetings/working-groups/roles/extension-wg-2024-06-07.md

```diff
@@ -0,0 +1,146 @@
+
+# Extensions WG 2024-06-07
+
+## Type erasure encoding
+
+We need to encode the extension type (for extension-aware compiler) and the underlying type (erasure and for non-extension-aware tools).
+
+We'll need to encode tuple names, dynamic bits, nullability, native integer for both types. 
+Ideally, this should be done in a way that leave any tools that understand current attribute/constructor.
+
+Options for encoding extension type:
+- generic attribute (pinging David to ask limitations/bugs on older frameworks)
+- attribute with typeof (may not refer to type parameters)
+- nested type
+- nested type with parallel method
+
+Decision: we'll spec out the parallel method in nested type design.
+
+### modopt
+```
+// source
+void M(E e)
+
+// codegen
+void M(C modopt(E) e)
+```
+
+```
+virtual void M2(C c)
+// other assembly
+override void M2(E e)
+```
+
+Con: binary break to change API from extension to underlying type and back
+Con: problem for override at language level (might be solvable)
+
+### Generic attribute

```

Comment:
Another con would be open generics support in generic attributes is not present yet, so a generic extension cannot use it. I'm adding the support in the runtime, so that generic extensions can use it in the future: https://github.com/dotnet/runtime/pull/103213